### DISCUSSION TITLE
new row violates row-level security policy for table xyz

### DISCUSSION THREAD

**Discussion by Kommune3** [supabase/supabase#2448] [Questions] [answered]
I'm currently struggling with a simple row level policy that checks on INSERTs for the users role. I took the policy from the provided template:

Only insert projects if user is authenticated
(role() = 'authenticated'::text)

I took the rule from the provided examples.

When a user inserts new data I still get:

Error Code 42501
Message: new row violates row-level security policy for table "Projects"

In Details:

I have a table "projects" with an uuid field and a project name. When I disable RLS all data is added without problems. Enabling it and adding the policy from above throws the error.

Any ideas?

**Comment by Kommune3** [comment, marked as answer]
Ok, I found the solution by myself:

You must have an Insert policy too and this should also be true. It's also possible to modify the insert command to:

await supabase.from('sample').insert({ key: 'value' }, { returning: 'minimal' })

Found the solution here: https://github.com/supabase/postgrest-js/issues/127

**Comment by thorwebdev** [comment]
Correct, by default you will need an INSERT and a SELECT policy as the INSERT query returns the newly added row which requires SELECT privileges. If you specify `returning: 'minimal'` the data is not being returned and therefore now SELECT policy is required then.

You can find an example of it here: https://github.com/supabase/supabase/blob/master/examples/react-todo-list/README.md#postgres-row-level-security
